typed-racket-lib/typed-racket/types/abbrev.rkt

```diff
@@ -34,7 +34,14 @@
 (define -thread-cell make-ThreadCell)
 (define -Promise make-Promise)
 (define -set make-Set)
+(define -Mutable-vec make-Mutable-Vector)
+(define -Immutable-vec make-Immutable-Vector)
+(define (make-Vector a) (Un (-Mutable-vec a) (-Immutable-vec a)))
 (define -vec make-Vector)
+(define (make-HeterogeneousVector ts) (Un (make-Immutable-HeterogeneousVector ts)
+                                          (make-Mutable-HeterogeneousVector ts)))
+(define (-Immutable-vec* . ts) (make-Immutable-HeterogeneousVector ts))
+(define (-Mutable-vec* . ts) (make-Mutable-HeterogeneousVector ts))
 (define (-vec* . ts) (make-HeterogeneousVector ts))
```

Comment:
Minor nitpick (sorry) - could the names for these abbreviated constructors (e.g. `-Immutable-vec`) be a little more consistent with the other abbreviated constructors? It looks like previously we've used either all lower case or camel case in the past (i.e. `-immutable-vec` or `-Immutable-Vec`), but I don't see any mixed-fix camel case for these sorts of abbreviations. Again, sorry for the minor nitpick =\